### PR TITLE
JDK 21 compatibility, version bumps and cleanup.

### PR DESCRIPTION
**Pull request by mbien**

### CODE REVIEW

**Comment by neilcsmith-net** [comment]
A useful start, although it also needs the JPMS and other aspects of the argline in the format required to pass to the launcher (eg. `-J--add-opens=...`.).  Not sure there's an easy way to generate one from the other, and two properties probably fine.  The archetype should ideally be set up from the start to write those into the conf file, or at least to the run action.

Having just migrated a large Ant platform application to Maven I have a few other thoughts on updating the archetype to work better out of the box.  I ended up copying some key aspects of structure from JJazzLab and Gephi.  Will see if I can find some time to follow up on this PR with a few other suggested changes.

File: pom.xml [hunk 1]
```diff
@@ -23,38 +23,43 @@ under the License.
...
-        <version>3</version>
+        <version>4</version>
     </parent>
     <modelVersion>4.0.0</modelVersion>
     <groupId>org.apache.netbeans.archetypes</groupId>
     <artifactId>netbeans-platform-app-archetype</artifactId>
     <version>1.24-SNAPSHOT</version>
     <packaging>maven-archetype</packaging>
-    <name>Apache Netbeans Maven Utilties - NetBeans Platform Application Archetype</name>
+    <name>Apache Netbeans Maven Utilities - NetBeans Platform Application Archetype</name>
```
Comment (mbien) [inline pom.xml:33]: btw can we make this shorter? This doesn't fit into the outline view of NB.

Comment (ebarboni) [reply pom.xml:33]: maybe Apache Netbeans Platform Application Maven Archetype ?

Comment (mbien) [reply pom.xml:33]: this is still a sentence and not a name :)

Comment (mbien) [reply pom.xml:33]: lets call it "NetBeans Platform Application Archetype"? rest is in the description

cc @neilcsmith-net

Comment (ebarboni) [reply pom.xml:33]: I think we "need" the Apache associated with NetBeans

Comment (mbien) [reply pom.xml]: changed it to "Apache NetBeans Platform Application Archetype" which is half of the original length.

**Comment by mbien** [comment]
```
            <plugins>
                <plugin>
                    <artifactId>maven-archetype-plugin</artifactId>
                    <version>3.2.1</version>
                </plugin>
            </plugins>
```
this seems to be https://mvnrepository.com/artifact/org.apache.maven.plugins/maven-archetype-plugin/3.2.1

is this correct? since there also is a maven-core plugin with the same `artifactId`.

**Comment by mbien** [comment]
merge?

**Comment by ebarboni** [comment]
maybe we should bump release there https://github.com/apache/netbeans-mavenutils-archetype-netbeans-platform-app-archetype/blob/90794b33c249abaaaa858693f01463c718004f72/src/main/resources/META-INF/maven/archetype-metadata.xml#L61C4-L61C46 to RELEASE220 or so

**Comment by mbien** [comment]
> maybe we should bump release there https://github.com/apache/netbeans-mavenutils-archetype-netbeans-platform-app-archetype/blob/90794b33c249abaaaa858693f01463c718004f72/src/main/resources/META-INF/maven/archetype-metadata.xml#L61C4-L61C46 to RELEASE220 or so

good catch! updated

**Review by ebarboni — APPROVED** [review]

**Pull request merged**